rust/chains/hyperlane-cosmos/src/providers/rpc.rs

```diff
@@ -86,55 +89,68 @@ impl CosmosWasmIndexer {
 }
 
 impl CosmosWasmIndexer {
-    #[instrument(level = "trace", err, skip(self))]
-    async fn tx_search(&self, query: Query, page: u32) -> ChainResult<TxSearchResponse> {
-        Ok(self
-            .provider
-            .rpc()
-            .tx_search(query, false, page, PAGINATION_LIMIT, Order::Ascending)
-            .await
-            .map_err(Into::<HyperlaneCosmosError>::into)?)
-    }
-
     // Iterate through all txs, filter out failed txs, find target events
     // in successful txs, and parse them.
     fn handle_txs<T>(
         &self,
-        txs: Vec<tx::Response>,
+        block: BlockResponse,
+        block_results: BlockResultsResponse,
         parser: for<'a> fn(&'a Vec<EventAttribute>) -> ChainResult<ParsedEvent<T>>,
-    ) -> ChainResult<impl Iterator<Item = (T, LogMeta)> + '_>
+    ) -> ChainResult<Vec<(T, LogMeta)>>
     where
         T: PartialEq + 'static,
     {
-        let logs_iter = txs
+        let Some(tx_results) = block_results.txs_results else {
+            return Ok(vec![]);
+        };
+
+        let tx_hashes: Vec<H256> = block
+            .clone()
```

Comment:
wondering if the clone here and line 132 are needed / if BlockResponse needs to be owned, iirc cosmos blocks can be MB in size. Not a big deal tho so feel free to leave as is

Reply:
all required sadly